From 5f5c2bbbe13a3309285efd358f54216014dd6a80 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Sun, 30 May 2021 17:19:41 +0300
Subject: [PATCH] [SW-43635] lkd: remove legacy protocol

Recently we moved from LKD protocol to COMMS
protocol. Therefore, is not required any more to
use the legacy protocol, because host is using
COMMS protocol.
This patch removes the legacy protocol
code. In case of identifying old legacy protocol,
an error print is displayed. In addition, dynamic
security, which opens sram space for image is not
required any more, because it was required only
with the legacy protocol, and therefore it is removed.

Change-Id: Id6f4e4f8971ec32c373cdcd3ce6f307efbecc296
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/preboot/module_lkd.c b/preboot/module_lkd.c
index b2402d0..72ae9b6 100644
--- a/preboot/module_lkd.c
+++ b/preboot/module_lkd.c
@@ -19,8 +19,6 @@
 #include "fit.h"
 
 struct lkd_status {
-	bool is_fit_checked;
-	bool is_fit_ready_ignored;
 	struct comms_command cmd;
 	struct comms_command wfc_cmd;
 	struct comms_command do_cmd;
@@ -40,8 +38,6 @@
 void module_lkd_setup(void)
 {
 	lkd_dev_specific_setup();
-	/* TODO: remove the below img open once we remove the old protocol */
-	preboot_security_dynamic_img_sram_open();
 	module_lkd_enable();
 }
 
@@ -66,55 +62,13 @@
 
 static enum cmd_check_res module_lkd_check_legacy(void)
 {
-	enum cmd_check_res res;
+	preboot_puts("legacy LKD not supported.\n");
 
-	switch (sts.cmd.val) {
-	case KMD_MSG_GOTO_WFE:
-		preboot_puts("LKD WFE received\n");
-		if (preboot_security_is_enabled()) {
-			cpu_boot_rsp_set(CPU_MSG_ERR);
-			preboot_puts("Security is enabled - WFE refused!\n");
-			res = NOT_FOUND;
-		} else {
-			/* WFE - we should clear the message/cmd register */
-			cpu_boot_cmd_clr_before_wfe();
-			cpu_boot_rsp_set(CPU_MSG_OK);
-			res = FOUND_NO_RETURN;
-		}
-		break;
-	case KMD_MSG_FIT_RDY:
-		if (sts.is_fit_ready_ignored)
-			return NOT_FOUND;
+	/* clear command register */
+	cpu_boot_cmd_clr_before_wfe();
+	sts.cmd.val = 0;
 
-		res = FOUND_SLOW_RETURN;
-		/* Tell LKD we got the message */
-		cpu_boot_rsp_set(CPU_MSG_OK);
-		preboot_puts("LKD FIT RDY received\n");
-		break;
-	case KMD_MSG_RST_DEV:
-		res = FOUND_NO_RETURN;
-		/* RST DEV - we should clear the message/cmd register */
-		cpu_boot_cmd_clr_before_wfe();
-		cpu_boot_rsp_set(CPU_MSG_OK);
-		preboot_puts("LKD RST DEV received\n");
-		break;
-	case KMD_MSG_SKIP_BMC:
-		cpu_boot_rsp_set(CPU_MSG_ERR);
-		preboot_puts("LKD SKIP BMC received\n");
-		/* no break */
-	default:
-		res = NOT_FOUND;
-	}
-
-	/*
-	 * Allows handling of any future FIT_RDY
-	 * command. Cause it means that the current command is any
-	 * but FIT_RDY. In this case there is no danger of a repeat
-	 * FIT_RDY command loop. It allows taking care of a FIT_RDY
-	 * at the FIT_RDY switch case and in the module run command
-	 */
-	sts.is_fit_ready_ignored = false;
-	return res;
+	return NOT_FOUND;
 }
 
 static int module_lkd_msg_reset_cause(struct lkd_fw_comms_msg *mcomms)
@@ -154,13 +108,8 @@
 {
 	struct lkd_fw_comms_msg *mcomms =
 		(struct lkd_fw_comms_msg *)FIT_IMG_BASE;
-	int err;
 
-	err = module_lkd_msg_process(mcomms);
-	if (err)
-		preboot_security_dynamic_img_sram_open();
-
-	return err;
+	return module_lkd_msg_process(mcomms);
 }
 
 static void module_lkd_reset_state(void)
@@ -243,22 +192,9 @@
 {
 	enum cmd_check_res res = NOT_FOUND;
 
-	/* we already have the FIT and it is good */
-	if (sts.is_fit_checked)
-		return FOUND_NO_RETURN;
-
 	sts.cmd.val = cpu_boot_cmd_get();
 	if (is_legacy_comms(sts.cmd.val))
 		return module_lkd_check_legacy();
-
-	/*
-	 * Allows handling of any future FIT_RDY
-	 * command. Cause it means that the current command is any
-	 * but FIT_RDY. In this case there is no danger of a repeat
-	 * FIT_RDY command loop. It allows taking care of a FIT_RDY
-	 * at the FIT_RDY switch case and in the module run command
-	 */
-	sts.is_fit_ready_ignored = false;
 
 	if (sts.cmd.cmd == sts.wfc_cmd.cmd)
 		return NOT_FOUND;
@@ -361,22 +297,6 @@
 	preboot_puts(" command\n");
 }
 
-static void module_lkd_run_legacy(void)
-{
-	int err;
-
-	preboot_security_dynamic_img_sram_close();
-
-	err = fit_load_images((void *)FIT_IMG_BASE);
-	if (err) {
-		sts.is_fit_ready_ignored = true;
-		preboot_security_dynamic_img_sram_open();
-		return;
-	}
-
-	sts.is_fit_checked = true;
-}
-
 __weak uintptr_t lkd_dev_spec_desc_offset(int size)
 {
 	return FIT_IMG_BASE;
@@ -412,10 +332,8 @@
 
 	/* shared mem doesn't contain fit image. */
 	err = module_lkd_msg_validate(FIT_IMG_BASE);
-	if (err) {
-		preboot_security_dynamic_img_sram_open();
+	if (err)
 		sts.exec_res = NOT_FOUND;
-	}
 
 	return err;
 }
@@ -423,8 +341,6 @@
 static int module_lkd_run_exec(void)
 {
 	/* Placeholder - empty for now */
-
-	/* on error do preboot_security_dynamic_img_open(); */
 
 	return 1;
 }
@@ -474,9 +390,6 @@
 {
 	int err = 0;
 	u32 offset = 0;
-
-	if (is_legacy_comms(sts.cmd.val))
-		return module_lkd_run_legacy();
 
 	/* depending on the current state machine, perform action */
 	switch (sts.do_cmd.cmd) {
@@ -556,28 +469,8 @@
 	wfe();
 }
 
-static void module_lkd_run_noret_legacy(void)
-{
-	switch (sts.cmd.val) {
-	case KMD_MSG_GOTO_WFE:
-		module_lkd_run_noret_wfe();
-		break;
-	case KMD_MSG_RST_DEV:
-		module_lkd_run_noret_reset();
-		break;
-	case KMD_MSG_FIT_RDY:
-		module_lkd_run_noret_exec();
-		/* no break */
-	default:
-		break;
-	}
-}
-
 void module_lkd_run_noret(void)
 {
-	if (is_legacy_comms(sts.cmd.val))
-		return module_lkd_run_noret_legacy();
-
 	/* we are not returning - clear comms status */
 	module_lkd_send_status(COMMS_STS_NOOP, COMMS_SRAM, 0);
 
